src/expr/node.cpp

```diff
@@ -22,7 +22,8 @@
 #include "base/output.h"
 #include "expr/attribute.h"
 
-#include "theory/quantifiers/term_database.h"
+// FIXME : can be removed if we properly handle #1205
+#include "theory/quantifiers/term_util.h"
```

Comment:
This introduces a new violation of the library build order. This should not go into master without a compelling reason.

What do we need from this header? We will need to refactor those parts out until we can respect the build order.

Reply:
The header is required for a fairly nasty hack to do SyGuS properly. The issue in the past is that we could not quantify on bound variables with function type.  Thus, the hack here relies on an attribute (SygusSynthFunVarListAttribute)  that effectively marks a function as being a bound variable.

This commit does not modify this hack (it just documents it a bit more).  I am waiting for the higher-order branch to merge so that this hack can be fixed.

Reply:
Naively it looks like a look of extra stuff is getting included into libexpr. Is it possible to move the attribute from src/theory then and push it into src/node in its own header.

Reply:
If you think this is high enough priority, I can fix this in a separate pull request.  On second thought, it might already be possible with the higher-order support that has been merged so far.

I'd prefer not to make major modifications to this pull request without confusing things, so I'd make a separate PR.

What do you think?

Reply:
I came up with a partial fix for #1205 in my latest commit that eliminates this dependency and shouldn't confuse this commit too much.  The solution was to have synthesis conjectures have free function variables (this is a better hack than the current one).